client/asset/eth/eth.go

```diff
@@ -789,25 +782,20 @@ func (*ExchangeWallet) SwapConfirmations(ctx context.Context, id dex.Bytes, cont
 }
 
 // sendToAddr sends funds from acct to addr.
-func (eth *ExchangeWallet) sendToAddr(addr common.Address, amt, gasPrice *big.Int) (common.Hash, error) {
-	tx := map[string]string{
-		"from":     fmt.Sprintf("0x%x", eth.acct.Address),
-		"to":       fmt.Sprintf("0x%x", addr),
-		"value":    fmt.Sprintf("0x%x", amt),
-		"gasPrice": fmt.Sprintf("0x%x", gasPrice),
+func (eth *ExchangeWallet) sendToAddr(addr common.Address, amt uint64) (*types.Transaction, error) {
+	txOpts, err := eth.node.txOpts(eth.ctx, amt, 0, defaultSendGasLimit)
+	if err != nil {
+		return nil, err
 	}
-	return eth.node.sendTransaction(eth.ctx, tx)
+	return eth.node.sendTransaction(eth.ctx, txOpts, addr, nil)
 }
 
 // Withdraw withdraws funds to the specified address. Value is gwei.
 //
 // TODO: Return the asset.Coin.
 // TODO: Subtract fees from the value.
-func (eth *ExchangeWallet) Withdraw(addr string, value, feeSuggestion uint64) (asset.Coin, error) {
-	bigVal := big.NewInt(0).SetUint64(value)
-	gweiFactorBig := big.NewInt(srveth.GweiFactor)
-	_, err := eth.sendToAddr(common.HexToAddress(addr),
-		bigVal.Mul(bigVal, gweiFactorBig), big.NewInt(0).SetUint64(defaultGasFee))
+func (eth *ExchangeWallet) Withdraw(addr string, value, _ uint64) (asset.Coin, error) {
+	_, err := eth.sendToAddr(common.HexToAddress(addr), value)
```

Comment:
We should revisit this at some point. If a user tries to withdraw their entire balance, it will fail and they'll lose the tx fees, I believe.

Reply:
Yikes, how does one sweep an ETH wallet completely, leaving nothing left if gas doesn't end up whatever was allocated to gas?

Reply:
Maybe the wallet won't allow the transaction to be sent the way I've described. I'm not certain.

Reply:
for ref: https://www.reddit.com/r/ethdev/comments/pits6s/sweep_all_eth_using_dynamic_fees_eip1559_and/
EIP-1559 does make this a headache if not impossible to do without leaving dust

Reply:
Haha, basically the way to sweep it is to give the dust to the miners.

For the original issue of the failing transaction, I would think that the node would have a check that refuses to submit a transaction where the `gasLimit*gasPrice + value > balance`, but I haven't tried it out yet.